src/Emulator/Cores/Arm-M/NVIC.cs

```diff
@@ -539,15 +548,18 @@ private int FindPendingInterrupt()
                         this.NoisyLog("IRQ {0} preempts {1}.", result, activeIRQs.Peek());
                     }
                 }
-                IRQ.Set(!PRIMASK && result != SpuriousInterrupt);
 
                 if(result != SpuriousInterrupt)
                 {
-                    maskedInterruptPresent.Set();
+                    maskedInterruptPresent = true;
+                    if(!PRIMASK)
+                    {
+                        IRQ.Set(true);
+                    }
                 }
                 else
                 {
-                    maskedInterruptPresent.Reset();
+                    maskedInterruptPresent = false;
```

Comment:
if we revert the `IRQ.Set` part, then this will become just `maskedInterruptPresent = result != SpuriousInterrupt;`